hushline/premium.py

```diff
@@ -0,0 +1,608 @@
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}", exc_info=True)
+                return jsonify(success=False), 400
+        else:
+            return jsonify(success=False), 400
+
+        user.stripe_subscription_cancel_at_period_end = True
+        db.session.add(user)
+        db.session.commit()
+
+        current_app.logger.info(
+            f"Autorenew disabled for subscription {user.stripe_subscription_id} for user {user.id}"
+        )
+
+        flash("Autorenew has been disabled.")
+        return jsonify(success=True)
+
+    @bp.route("/enable-autorenew", methods=["POST"])
+    @authentication_required
+    def enable_autorenew() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                stripe.Subscription.modify(user.stripe_subscription_id, cancel_at_period_end=False)
+                user.stripe_subscription_cancel_at_period_end = False
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Autorenew enabled for subscription {user.stripe_subscription_id} for user {user.id}"  # noqa: E501
+                )
+
+                flash("Autorenew has been enabled.")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}", exc_info=True)
+                return jsonify(success=False), 400
+
+        return jsonify(success=False), 400
+
+    @bp.route("/cancel", methods=["POST"])
+    @authentication_required
+    def cancel() -> Response | str | Tuple[Response | str, int]:
+        user = db.session.get(User, session.get("user_id"))
+        if not user:
+            session.clear()
+            return redirect(url_for("login"))
+
+        if user.stripe_subscription_id:
+            try:
+                # Cancel the subscription
+                stripe.Subscription.delete(user.stripe_subscription_id)
+
+                # Downgrade the user (the subscription ID will get removed in the webhook)
+                user.tier_id = FREE_TIER
+                db.session.add(user)
+                db.session.commit()
+
+                current_app.logger.info(
+                    f"Subscription {user.stripe_subscription_id} canceled for user {user.id}"
+                )
+
+                flash("💔 Sorry to see you go!")
+                return jsonify(success=True)
+            except stripe._error.StripeError as e:
+                current_app.logger.error(f"Stripe error: {e}", exc_info=True)
+                return jsonify(success=False), 400
+
+        return jsonify(success=False), 400
+
+    @bp.route("/status.json", methods=["GET"])
```

Comment:
Nit: `methods` are unnecessary when the endpoint is only `GET`

Reply:
Fixed in 40b1ca102ea381b325f06f6a1ec5ceb61c4228a2